BaseClasses.py

```diff
@@ -969,6 +969,28 @@ def add_locations(self, locations: Dict[str, Optional[int]],
         for location, address in locations.items():
             self.locations.append(location_type(self.player, location, address, self))
 
+    def add_event(self, location_name: str, item_name: Optional[str] = None,
+                  rule: Optional[Callable[[CollectionState], bool]] = None) -> Item:
+        """
+        Adds an event location/item pair to a region.
+
+        :param location_name: Name for the event location.
+        :param item_name: Name for the event item. If not provided, defaults to location_name.
+        :param rule: Callable to determine access for this event location within its region.
+        """
+        if item_name is None:
+            item_name = location_name
+        event_item = Item(item_name, ItemClassification.progression, None, self.player)
+
+        event_location = Location(self.player, location_name, None, self)
```

Comment:
Hm, this does cause the event item/location to be created with the generic Item/Location classes instead of the game-specific subclasses. My guess is that doesn't matter for an event, since the only user-facing or even developer-facing behavior I know of where the game-specific subclasses matter is allowing other games to show the name of the game an item is from, which clearly doesn't matter for events. How sure are we that it has no other effects?

Reply:
Yeah I was thinking about that.

Two considerations I had:

**Precedent:** In the `region.add_locations` function, you can provide a location class. **Optionally.** It will let you create locations with the base Location class.
**Alternatives:** `location.player` already exists, and it's the much better way to figure out what player a location belongs to than trying to check the type of the location class. If you then need the game, `multiworld.worlds[player].game`. In fact, I would actively discourage trying to determine the game a location came from from its class, that just seems bad and janky and unnecessary

Reply:
One compelling reason this should use the game specific classes is if custom collect is needed on the events (HK would need this, e.g. in the case of vanilla charms)